### PR TITLE
hotfix/fix post controller

### LINKED ISSUE #34: Complete crud operations for /posts endpoint
GET, POST, PUT, and DELETE methods for the post controller

### PR DESCRIPTION
**Pull request by themellowpea**
### TL;DR

- I updated the posts table and adjusted the sequence generation.
- Implemented full CRUD functionality in PostService and PostController.
- Configured DTO objects and integrated MapStruct for entity mapping.
- Comprehensive unit tests for the PostController were added.

#### Description

This PR encapsulates a series of changes and enhancements to implement and test the /posts endpoint fully. Below is a detailed breakdown of the changes:

#### Commits Overview:

1. update posts table names:
- Refined the posts table structure by updating column names for better clarity and alignment with project conventions.
2. use LocalDateTime:
- Updated timestamp fields to use LocalDateTime for accurate date-time handling in the Post entity.
3. change bigint to long:
- Changed column types from bigint to long where appropriate, ensuring consistency with Java data types.
4. configure PostService, add all CRUD endpoints:
- Implemented the full range of CRUD operations within the PostService and exposed them via PostController.
- Introduced methods to create, read, update, and delete posts.
5. update PostController, comment test code out:
- Updated the PostController to integrate the new CRUD operations.
- Temporarily commented out some test code for further refinement and integration.
6. setup DTO objects and MapStruct mapper interface, update service layer:
- Created PostRequestDTO and PostResponseDTO for clean data handling.
- Configured MapStruct for automatic mapping between DTOs and entities.
- Refined the service layer to utilize these DTOs effectively.
7. add tests for PostController and update MapStruct and Lombok configurations:
- Developed comprehensive unit tests for the PostController to ensure robust functionality.
- Updated MapStruct and Lombok configurations to support the new DTO mappings and service logic.
8. update posts table and fix sequence generation:
- Finalized the posts table updates and fixed issues related to sequence generation, ensuring smooth operation during data insertion and retrieval.

### CODE REVIEW

**Review by trevorlana — APPROVED** [review]
Looks good

File: src/main/java/com/rev_connect_api/config/SecurityConfig.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
             .authorizeHttpRequests(authorize -> authorize
                 .requestMatchers("/auth/login", "/users/register", "/h2-console/**", "/attachments/**").permitAll() // allow login and register to everyone
                 .requestMatchers("/users/**", "/posts/**", "/comments/**", "/likes/**", "/profile/?*").hasAnyRole("USER", "ADMIN") // USER or ADMIN roles can access /users/**
-                .anyRequest().authenticated()) // all other requests require authentication
+                .anyRequest().anonymous()) // all other requests require authentication
```
Comment (revature-ben-m) [inline src/main/java/com/rev_connect_api/config/SecurityConfig.java:33]: https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurer.AuthorizedUrl.html

Did you mean to change this back? Based on the comment I assumed you wanted to require full authentication for all endpoints not specified. My understanding is that this allows users that do not have an auth token.

Comment (themellowpea) [reply src/main/java/com/rev_connect_api/config/SecurityConfig.java:33]: Any request, not specified is anonymous, so unauthenticated. This returns the correct 403 on attempts to reach endpoints, and the ones specified will be permitted if they have the correct jwt token/roles

Comment (revature-ben-m) [reply src/main/java/com/rev_connect_api/config/SecurityConfig.java:33]: I don't think this is the case. Sending a get request to http://localhost:8080/api/notAnEndpoint returns a 404 when set to anonymous but returns a 403 when set to authenticated.

Comment (revature-ben-m) [reply src/main/java/com/rev_connect_api/config/SecurityConfig.java:33]: In the grand scheme of things I think we're fine to merge this for now in either state.

**Review by revature-ben-m — APPROVED** [review]

File: src/main/java/com/rev_connect_api/config/SecurityConfig.java (hunk 1 above)
Comment (themellowpea) [reply src/main/java/com/rev_connect_api/config/SecurityConfig.java:33]: that's correct. It should send a 404 since that endpoint doesn't exist. setting to authenticated is forcing authentication on endpoints the server doesn't support.
